XR_APILAYER_MBUCCHIA_toolkit/imageprocess.cpp

```diff
@@ -86,12 +90,40 @@ namespace {
             // TODO: We can use an IShaderBuffer cache per swapchain and avoid this every frame.
             m_cbParams->uploadData(config, sizeof(*config));
 
-            const auto usePostProcess = m_mode == PostProcessType::On;
-            m_device->setShader(m_shaders[usePostProcess], SamplerType::LinearClamp);
+           
+            if (m_caCorrectionType != PostProcessCACorrectionType::Off) {
+                const auto outputInfo = output->getInfo();
+                if (textures.empty() || textures[0]->getInfo().width != outputInfo.width ||
+                    textures[0]->getInfo().height != outputInfo.height) {
+                    textures.clear();
+                    auto createInfo = outputInfo;
+
+                    // Good balance between visuals and performance.
+                    createInfo.format = m_device->getTextureFormat(TextureFormat::R16G16B16A16_UNORM);
+
+                    createInfo.createFlags |= D3D11_BIND_SHADER_RESOURCE;
```

Comment:
You don't need this. These are valid OpenXR swapchain create flags: https://registry.khronos.org/OpenXR/specs/1.0/man/html/XrSwapchainCreateFlagBits.html

`XR_SWAPCHAIN_USAGE_SAMPLED_BIT` below is doing what you want.